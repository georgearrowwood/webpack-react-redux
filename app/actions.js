import productsApi from './api/products'

export const addProduct = (title) => (dispatch, getState) => {
  return {
    type: 'ADD_PRODUCT',
    title
  }
}

export const requestProducts = () => ({
  type: 'REQUEST_PRODUCTS'
})

export const receiveProducts = (data) => ({
  type: 'RECEIVE_PRODUCTS',
  products: data
})

export const fetchProducts = () => (dispatch, getState) => {
  const state = getState()
  if (state.isFetching) return false
  dispatch(requestProducts())
  productsApi.getList()
    .then((data) => {
      dispatch(receiveProducts(data.data.products))
    })
}
