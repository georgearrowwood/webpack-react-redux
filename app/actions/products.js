import dispatcher from '../utils/dispatcher';
import productConstants from '../constants/products';
import productsApi from '../api/products';

export default {
  /**
   * @param  {string} text
   */
  create: function(product) {
    if (product && product.title) {
      productsApi.create(product)
        .then(result => {
          if (result.data.success) {
            this.fetchList();
          }
        })
    }
  },

  fetchList: function() {
    productsApi.getList()
      .then(result => {
        dispatcher.dispatch({
          actionType: productConstants.PRODUCTS_LIST_DATA_RECEIVED,
          products: result.data.products
        });
      })
  },

  /**
   * @param  {integer} id
   */
  delete: function(id) {
    console.log('d1', id);
    if (Number.isInteger(id)) {
      productsApi.delete(id)
        .then(result => {
          if (result.data.success) {
            this.fetchList();
          }
        })
    }
  }

};
