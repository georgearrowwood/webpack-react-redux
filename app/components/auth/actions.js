import request from '../../modules/request';

export const loginAction = ({ login, password }, history) => (dispatch) => {
    console.log('6', login);
      
    request.post('/api/login', { login, password })
      .then(res => {   
        console.log('res log', res);
        
        dispatch({ type: 'AUTHENTICATED' });
        /* eslint-env browser */
        localStorage.setItem('user-token', res.data.token);
        // history.push('/');
        history.goBack();
      })  
      .catch ((error) => {
        console.log('err log', error);
        dispatch({
          type: 'AUTHENTICATION_ERROR',
          payload: 'Invalid email or password',
        });
      });
};

export const as = 1;

