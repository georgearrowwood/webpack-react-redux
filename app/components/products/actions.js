import services from './services';

export const requestProducts = () => ({
  type: 'REQUEST_PRODUCTS',
});

export const receiveProducts = data => ({
  type: 'RECEIVE_PRODUCTS',
  products: data,
});

export const fetchProducts = () => (dispatch, getState) => {
  const state = getState();
  if (state.isFetching) return false;

  dispatch(requestProducts());
  return services().getList()
    .then(data => dispatch(receiveProducts(data.data.products)));
};

export const addProduct = data => (dispatch) => {
  services().addOne(data.title)
    .then(() => {
      dispatch(fetchProducts());
    });
};

export const deleteProduct = id => dispatch =>
  services().deleteOne(id)
    .then(() => {
      dispatch(fetchProducts());
    });
