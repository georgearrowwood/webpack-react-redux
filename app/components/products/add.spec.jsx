import React from 'react';
import { mount } from 'enzyme';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import ProductsAdd from './add';
import * as productsActions from './actions';

const store = createStore(combineReducers({ form: formReducer }), applyMiddleware(thunk));

let wrapper;
let Form;
let input;

beforeAll(() => {
  wrapper = mount(
    <Provider store={store}><ProductsAdd /></Provider>);
  Form = wrapper.find('ProductAddForm');
  input = Form.find('input#product-title');
});

describe('Product add form', () => {
  it('Renders products add form elements', () => {
    expect(wrapper).toMatchSnapshot();
    expect(typeof Form.prop('handleSubmit')).toEqual('function');
    expect(typeof input).toEqual('object');
    expect(typeof Form.find('button')).toEqual('object');
  });

  it('Simulates text entry', () => {
    jest.spyOn(productsActions, 'addProduct').mockImplementation(() => jest.fn());
    input.simulate('change', { target: { value: 'asd' } });
    Form.simulate('submit');
    expect(productsActions.addProduct.mock.calls.length).toEqual(1);
    expect(typeof productsActions.addProduct.mock.calls[0][0]).toBe('object');
    expect(productsActions.addProduct.mock.calls[0][0].title).toBe('asd');
    productsActions.addProduct.mockRestore();
  });

  
});
