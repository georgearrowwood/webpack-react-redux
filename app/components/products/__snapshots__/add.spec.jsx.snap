// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product add form Renders products add form elements 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReduxForm>
    <Connect(Form(ProductAddForm))
      destroyOnUnmount={true}
      enableReinitialize={false}
      forceUnregisterOnUnmount={false}
      form="productAdd"
      getFormState={[Function]}
      initialValues={undefined}
      keepDirtyOnReinitialize={false}
      persistentSubmitErrors={false}
      pure={true}
      shouldAsyncValidate={[Function]}
      shouldError={[Function]}
      shouldValidate={[Function]}
      shouldWarn={[Function]}
      touchOnBlur={true}
      touchOnChange={false}
    >
      <Form(ProductAddForm)
        anyTouched={false}
        array={
          Object {
            "insert": [Function],
            "move": [Function],
            "pop": [Function],
            "push": [Function],
            "remove": [Function],
            "removeAll": [Function],
            "shift": [Function],
            "splice": [Function],
            "swap": [Function],
            "unshift": [Function],
          }
        }
        arrayInsert={[Function]}
        arrayMove={[Function]}
        arrayPop={[Function]}
        arrayPush={[Function]}
        arrayRemove={[Function]}
        arrayRemoveAll={[Function]}
        arrayShift={[Function]}
        arraySplice={[Function]}
        arraySwap={[Function]}
        arrayUnshift={[Function]}
        asyncErrors={undefined}
        asyncValidating={false}
        autofill={[Function]}
        blur={[Function]}
        change={[Function]}
        clearAsyncError={[Function]}
        clearSubmit={[Function]}
        clearSubmitErrors={[Function]}
        destroy={[Function]}
        destroyOnUnmount={true}
        dirty={false}
        dispatch={[Function]}
        enableReinitialize={false}
        error={undefined}
        focus={[Function]}
        forceUnregisterOnUnmount={false}
        form="productAdd"
        getFormState={[Function]}
        initialValues={undefined}
        initialize={[Function]}
        initialized={false}
        invalid={false}
        keepDirtyOnReinitialize={false}
        persistentSubmitErrors={false}
        pristine={true}
        pure={true}
        registerField={[Function]}
        registeredFields={
          Object {
            "title": Object {
              "count": 1,
              "name": "title",
              "type": "Field",
            },
          }
        }
        reset={[Function]}
        setSubmitFailed={[Function]}
        setSubmitSucceeded={[Function]}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        startAsyncValidation={[Function]}
        startSubmit={[Function]}
        stopAsyncValidation={[Function]}
        stopSubmit={[Function]}
        submit={[Function]}
        submitFailed={false}
        submitSucceeded={false}
        submitting={false}
        syncErrors={Object {}}
        syncWarnings={Object {}}
        touch={[Function]}
        touchOnBlur={true}
        touchOnChange={false}
        triggerSubmit={undefined}
        unregisterField={[Function]}
        untouch={[Function]}
        updateSyncErrors={[Function]}
        updateSyncWarnings={[Function]}
        valid={true}
        validExceptSubmit={true}
        values={Object {}}
        warning={undefined}
      >
        <ProductAddForm
          anyTouched={false}
          array={
            Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            }
          }
          asyncValidate={[Function]}
          asyncValidating={false}
          autofill={[Function]}
          blur={[Function]}
          change={[Function]}
          clearAsyncError={[Function]}
          clearSubmit={[Function]}
          clearSubmitErrors={[Function]}
          destroy={[Function]}
          dirty={false}
          dispatch={[Function]}
          error={undefined}
          form="productAdd"
          handleSubmit={[Function]}
          initialValues={undefined}
          initialize={[Function]}
          initialized={false}
          invalid={false}
          pristine={true}
          pure={true}
          reset={[Function]}
          submit={[Function]}
          submitFailed={false}
          submitSucceeded={false}
          submitting={false}
          touch={[Function]}
          triggerSubmit={undefined}
          untouch={[Function]}
          valid={true}
          warning={undefined}
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <label
                htmlFor="product-title"
              >
                <span>
                  Product title
                </span>
                <div>
                  <Field
                    component="input"
                    id="product-title"
                    name="title"
                    placeholder="Product Title"
                    type="text"
                  >
                    <Connect(ConnectedField)
                      _reduxForm={
                        Object {
                          "anyTouched": false,
                          "array": Object {
                            "insert": [Function],
                            "move": [Function],
                            "pop": [Function],
                            "push": [Function],
                            "remove": [Function],
                            "removeAll": [Function],
                            "shift": [Function],
                            "splice": [Function],
                            "swap": [Function],
                            "unshift": [Function],
                          },
                          "arrayInsert": [Function],
                          "arrayMove": [Function],
                          "arrayPop": [Function],
                          "arrayPush": [Function],
                          "arrayRemove": [Function],
                          "arrayRemoveAll": [Function],
                          "arrayShift": [Function],
                          "arraySplice": [Function],
                          "arraySwap": [Function],
                          "arrayUnshift": [Function],
                          "asyncErrors": undefined,
                          "asyncValidate": [Function],
                          "asyncValidating": false,
                          "autofill": [Function],
                          "blur": [Function],
                          "change": [Function],
                          "clearAsyncError": [Function],
                          "clearSubmit": [Function],
                          "clearSubmitErrors": [Function],
                          "destroy": [Function],
                          "destroyOnUnmount": true,
                          "dirty": false,
                          "dispatch": [Function],
                          "enableReinitialize": false,
                          "error": undefined,
                          "focus": [Function],
                          "forceUnregisterOnUnmount": false,
                          "form": "productAdd",
                          "getFormState": [Function],
                          "getValues": [Function],
                          "initialValues": undefined,
                          "initialize": [Function],
                          "initialized": false,
                          "invalid": false,
                          "keepDirtyOnReinitialize": false,
                          "persistentSubmitErrors": false,
                          "pristine": true,
                          "pure": true,
                          "register": [Function],
                          "registerField": [Function],
                          "registerInnerOnSubmit": [Function],
                          "registeredFields": undefined,
                          "reset": [Function],
                          "sectionPrefix": undefined,
                          "setSubmitFailed": [Function],
                          "setSubmitSucceeded": [Function],
                          "shouldAsyncValidate": [Function],
                          "shouldError": [Function],
                          "shouldValidate": [Function],
                          "shouldWarn": [Function],
                          "startAsyncValidation": [Function],
                          "startSubmit": [Function],
                          "stopAsyncValidation": [Function],
                          "stopSubmit": [Function],
                          "submit": [Function],
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "submitting": false,
                          "syncErrors": Object {},
                          "syncWarnings": Object {},
                          "touch": [Function],
                          "touchOnBlur": true,
                          "touchOnChange": false,
                          "triggerSubmit": undefined,
                          "unregister": [Function],
                          "unregisterField": [Function],
                          "untouch": [Function],
                          "updateSyncErrors": [Function],
                          "updateSyncWarnings": [Function],
                          "valid": true,
                          "validExceptSubmit": true,
                          "values": Object {},
                          "warning": undefined,
                        }
                      }
                      component="input"
                      id="product-title"
                      name="title"
                      normalize={[Function]}
                      placeholder="Product Title"
                      type="text"
                    >
                      <ConnectedField
                        _reduxForm={
                          Object {
                            "anyTouched": false,
                            "array": Object {
                              "insert": [Function],
                              "move": [Function],
                              "pop": [Function],
                              "push": [Function],
                              "remove": [Function],
                              "removeAll": [Function],
                              "shift": [Function],
                              "splice": [Function],
                              "swap": [Function],
                              "unshift": [Function],
                            },
                            "arrayInsert": [Function],
                            "arrayMove": [Function],
                            "arrayPop": [Function],
                            "arrayPush": [Function],
                            "arrayRemove": [Function],
                            "arrayRemoveAll": [Function],
                            "arrayShift": [Function],
                            "arraySplice": [Function],
                            "arraySwap": [Function],
                            "arrayUnshift": [Function],
                            "asyncErrors": undefined,
                            "asyncValidate": [Function],
                            "asyncValidating": false,
                            "autofill": [Function],
                            "blur": [Function],
                            "change": [Function],
                            "clearAsyncError": [Function],
                            "clearSubmit": [Function],
                            "clearSubmitErrors": [Function],
                            "destroy": [Function],
                            "destroyOnUnmount": true,
                            "dirty": false,
                            "dispatch": [Function],
                            "enableReinitialize": false,
                            "error": undefined,
                            "focus": [Function],
                            "forceUnregisterOnUnmount": false,
                            "form": "productAdd",
                            "getFormState": [Function],
                            "getValues": [Function],
                            "initialValues": undefined,
                            "initialize": [Function],
                            "initialized": false,
                            "invalid": false,
                            "keepDirtyOnReinitialize": false,
                            "persistentSubmitErrors": false,
                            "pristine": true,
                            "pure": true,
                            "register": [Function],
                            "registerField": [Function],
                            "registerInnerOnSubmit": [Function],
                            "registeredFields": undefined,
                            "reset": [Function],
                            "sectionPrefix": undefined,
                            "setSubmitFailed": [Function],
                            "setSubmitSucceeded": [Function],
                            "shouldAsyncValidate": [Function],
                            "shouldError": [Function],
                            "shouldValidate": [Function],
                            "shouldWarn": [Function],
                            "startAsyncValidation": [Function],
                            "startSubmit": [Function],
                            "stopAsyncValidation": [Function],
                            "stopSubmit": [Function],
                            "submit": [Function],
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "submitting": false,
                            "syncErrors": Object {},
                            "syncWarnings": Object {},
                            "touch": [Function],
                            "touchOnBlur": true,
                            "touchOnChange": false,
                            "triggerSubmit": undefined,
                            "unregister": [Function],
                            "unregisterField": [Function],
                            "untouch": [Function],
                            "updateSyncErrors": [Function],
                            "updateSyncWarnings": [Function],
                            "valid": true,
                            "validExceptSubmit": true,
                            "values": Object {},
                            "warning": undefined,
                          }
                        }
                        _value={undefined}
                        asyncError={undefined}
                        asyncValidating={false}
                        component="input"
                        dirty={false}
                        dispatch={[Function]}
                        id="product-title"
                        initial={undefined}
                        name="title"
                        normalize={[Function]}
                        placeholder="Product Title"
                        pristine={true}
                        state={undefined}
                        submitError={undefined}
                        submitFailed={undefined}
                        submitting={undefined}
                        syncError={undefined}
                        syncWarning={undefined}
                        type="text"
                        value={undefined}
                      >
                        <input
                          id="product-title"
                          name="title"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onDragStart={[Function]}
                          onDrop={[Function]}
                          onFocus={[Function]}
                          placeholder="Product Title"
                          type="text"
                          value=""
                        />
                      </ConnectedField>
                    </Connect(ConnectedField)>
                  </Field>
                </div>
              </label>
            </div>
            <div>
              <button
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </ProductAddForm>
      </Form(ProductAddForm)>
    </Connect(Form(ProductAddForm))>
  </ReduxForm>
</Provider>
`;
